### PR TITLE
LMB-376: POC ldes consumer for mandatenbeheer

### PR DESCRIPTION
**Pull request by JonasVanHoof**
⚠️ DO NOT MERGE ⚠️

We want to sync the data of loket with the mandatenbheer application. This will be done by consuming the LMB LDES feed. To make sure that everything in this POC is understandable a readme file is added in the docs directory. Definitly read it!

Testing with LMB is possible when checking out on https://github.com/lblod/app-lokaal-mandatenbeheer/pull/119

### CODE REVIEW

**Review by Rahien — CHANGES_REQUESTED** [review]

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -20,15 +20,15 @@ services:
...
+      - "91:80"
     restart: "no"
   dispatcher:
     restart: "no"
   database:
     restart: "no"
   virtuoso:
     ports:
-      - "8890:8890"
+      - "8880:8890"
```
[the remark this answers is not in the document]
Comment (JonasVanHoof) [reply docker-compose.dev.yml:31]: Hehe but I was thinking this is how it works now. Do we change our database port when we want to test run it?

File: docker-compose.yml [hunk 2]
```diff
@@ -709,3 +709,27 @@ services:
...
     logging: *default-logging
+  lmb-fractie-consumer:
+    image: redpencil/ldes-consumer:0.8.0-rc1
+    environment:
+      # LDES_STREAM: "http://ldes-backend/fractie" TODO: LDES of mandatenbeheer
+      # LDES_ENDPOINT_VIEW: "http://ldes-backend/fractie/1" TODO: LDES of of mandatenbeheer
+      CRON_PATTERN: "*/1 * * * *" # run the job every 1 minute
+      REPLACE_VERSIONS: true 
+      MU_SPARQL_ENDPOINT: 'http://database:8890/sparql'
+      MU_APPLICATION_GRAPH: "http://mu.semte.ch/graphs/mandaten-staging"
```
[the remark this answers is not in the document]
Comment (JonasVanHoof) [reply docker-compose.yml:720]: Need to add `PERSIST_STATE: "true"` here

**Review by Rahien — APPROVED** [review]

File: docs/poc-sync-mandatenbeheer-entities.md
Comment (cecemel) [inline docs/poc-sync-mandatenbeheer-entities.md:1]: Can this information be merged in the `README.md` file?

File: docker-compose.yml [hunk 3]
```diff
@@ -665,3 +665,30 @@ services:
...
+      PERSIST_STATE: "true"
+    labels:
+        - "logging=true"
+    restart: always
+    logging: *default-logging
+  lmb-upsert-ldes-entity:
+    image: lblod/ldes-make-object-service:latest
```
Comment (cecemel) [inline docker-compose.yml:685]: Is is possible to provide a decent tag? `Latest` is often confusing.

File: docker-compose.dev.yml [hunk 4]
```diff
@@ -150,3 +150,14 @@ services:
     restart: "no"
   berichtencentrum-melding:
     restart: "no"
+  lmb-ldes-consumer:
+    environment:
+      CRON_PATTERN: "*/15 * * * * *" # run the job every 15 seconds
+    networks:
```
Comment (cecemel) [inline docker-compose.dev.yml:156]: can this be removed?

Comment (JonasVanHoof) [reply docker-compose.dev.yml:156]: Sure, could add it to the overwrite if needed

File: docker-compose.yml (hunk 3 above)
Comment (JonasVanHoof) [reply docker-compose.yml:685]: Need to look in what tag to use than 👌🏻  https://hub.docker.com/r/lblod/ldes-make-object-service/tags

File: docs/poc-sync-mandatenbeheer-entities.md
Comment (JonasVanHoof) [reply docs/poc-sync-mandatenbeheer-entities.md:1]: Questioned Karel if the information is correct @Rahien

File: docker-compose.yml [hunk 5]
```diff
@@ -665,3 +665,30 @@ services:
...
+    depends_on:
+      - lmb-ldes-consumer
+    environment:
+      LANDING_ZONE_GRAPH: "http://mu.semte.ch/graphs/mandaten-staging"
+      TARGET_GRAPH: "http://mu.semte.ch/graphs/mandaten"
```
Comment (cecemel) [inline docker-compose.yml:690]: can this be renamed, to e.g. `http://mu.semte.ch/graphs/lmb/mandaten`? Cause we have many types of mandaten in  loket

File: docker-compose.yml [hunk 6]
```diff
@@ -665,3 +665,30 @@ services:
...
+    labels:
+        - "logging=true"
+    restart: always
+    logging: *default-logging
+  lmb-upsert-ldes-entity:
+    image: lblod/ldes-make-object-service:latest
+    depends_on:
+      - lmb-ldes-consumer
+    environment:
+      LANDING_ZONE_GRAPH: "http://mu.semte.ch/graphs/mandaten-staging"
```
Comment (cecemel) [inline docker-compose.yml:689]: idem as next comment :-)

File: docs/poc-sync-mandatenbeheer-entities.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+When the triple is passed on the `/post` method that versioned triple will be transformed to the `isVersionOf` to create or update the actual entity. This entity is than upserted to the loket database (`TARGET_GRAPH`).
+
+### Deltanotifier
+
+No extra confiruation for this service was done only writing the rule to make sure that only the triples from our staging graph (`LANDING_ZONE_GRAPH`) are being processed.
+
+### LMB application
+
+Not sure how we are going to include it in this application. TODO:
```
Comment (JonasVanHoof) [inline docs/poc-sync-mandatenbeheer-entities.md:30]: remove* our application will be linked through the module card

File: docs/poc-sync-mandatenbeheer-entities.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# Sync mandatenbeheer entities (POC)
+
+Lokaal Mandatenbeheer ([LMB](https://github.com/lblod/app-lokaal-mandatenbeheer)) will manage all madatarissen. These mandatarissen are build up from different entities than can be edited. (Fractie,..) To keep Loket in sync with all these changes we want to consume the LDES dat from the LMB LDES feed.
+
+## Services
+
+- LDES consumer (redpencil/ldes-consumer:0.8.0-rc1)
+- Upsert entities (lblod/ldes-make-object-service)
+- Deltanotifier (cecemel/delta-notifier:0.2.0-beta.2)
+- LMB application
```
Comment (JonasVanHoof) [inline docs/poc-sync-mandatenbeheer-entities.md:10]: also remove*

File: docker-compose.yml [hunk 9]
```diff
@@ -665,3 +665,30 @@ services:
...
+    image: redpencil/ldes-consumer:0.8.0-rc1
+    depends_on:
+      - database
+    environment:
+      LDES_STREAM: "https://mandatenbeheer.lblod.info/streams/ldes/public"
+      LDES_ENDPOINT_VIEW: "https://mandatenbeheer.lblod.info/streams/ldes/public/1"
+      CRON_PATTERN: "*/1 * * * *" # run the job every 1 minute
+      REPLACE_VERSIONS: true 
+      MU_SPARQL_ENDPOINT: 'http://database:8890/sparql'
+      MU_APPLICATION_GRAPH: "http://mu.semte.ch/graphs/mandaten-staging"
```
Comment (bfidlers) [inline docker-compose.yml:678]: I think this should also be "http://mu.semte.ch/graphs/lmb/mandaten-staging"

File: config/delta/upsert-mandatenbeheer-entity.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+export default [
+  {
+    match: {
+      graph: {
+        type: 'uri',
+        value: 'http://mu.semte.ch/graphs/mandaten-staging',
```
Comment (bfidlers) [inline config/delta/upsert-mandatenbeheer-entity.js:6]: This should be "http://mu.semte.ch/graphs/lmb/mandaten-staging"

File: docker-compose.dev.yml [hunk 11]
```diff
@@ -150,3 +150,13 @@ services:
     restart: "no"
   berichtencentrum-melding:
     restart: "no"
+  lmb-ldes-consumer:
+    networks:
+      - default
+      - debug
+  lmb-upsert-ldes-entity:
+    image: lblod/ldes-make-object-service:latest
```
Comment (cecemel) [inline docker-compose.dev.yml:158]: can this be removed?

Comment (bfidlers) [reply docker-compose.dev.yml:158]: yeah we should update the image when a new version has been released, but for now we can just use the image used in de docker-compose.yml

**Pull request closed without merging**